sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/schemas/Sgv2IndexesResourceImpl.java

```diff
@@ -0,0 +1,177 @@
+import javax.ws.rs.core.Response;
+
+@Path("/v2/schemas/keyspaces/{keyspaceName}/tables/{tableName}/indexes")
+@Produces(MediaType.APPLICATION_JSON)
+@Singleton
+public class Sgv2IndexesResourceImpl extends ResourceBase implements Sgv2IndexesResourceApi {
+
+  @Inject private GrpcClientFactory grpcFactory;
+
+  @Override
+  public Response getAllIndexesForTable(
+      String token, String keyspaceName, String tableName, HttpServletRequest request) {
+    if (isAuthTokenInvalid(token)) {
+      return invalidTokenFailure();
+    }
+    return Sgv2RequestHandler.handleMainOperation(
+        () -> {
+          if (isStringEmpty(keyspaceName)) {
+            return jaxrsBadRequestError("keyspaceName must be provided").build();
+          }
+          if (isStringEmpty(tableName)) {
+            return jaxrsBadRequestError("tableName must be provided").build();
+          }
+
+          StargateGrpc.StargateBlockingStub blockingStub =
+              grpcFactory
+                  .constructBlockingStub()
+                  .withCallCredentials(new StargateBearerToken(token));
+
+          BridgeSchemaClient.create(blockingStub).findTable(keyspaceName, tableName);
+
+          String cql =
+              new QueryBuilder()
+                  .select()
+                  .from("system_schema", "indexes")
+                  .where("keyspace_name", Predicate.EQ, keyspaceName)
+                  .where("table_name", Predicate.EQ, tableName)
+                  .build();
+          return fetchRows(blockingStub, -1, null, true, cql, null);
+        });
+  }
+
+  public Response addIndex(
+      String token,
+      final String keyspaceName,
+      final String tableName,
+      final Sgv2IndexAddRequest indexAdd,
+      HttpServletRequest request) {
+    if (isAuthTokenInvalid(token)) {
+      return invalidTokenFailure();
+    }
+    return Sgv2RequestHandler.handleMainOperation(
+        () -> {
+          if (isStringEmpty(keyspaceName)) {
+            return jaxrsBadRequestError("keyspaceName must be provided").build();
+          }
+          if (isStringEmpty(tableName)) {
+            return jaxrsBadRequestError("tableName must be provided").build();
+          }
+
+          String columnName = indexAdd.getColumn();
+          if (isStringEmpty(columnName)) {
+            return jaxrsBadRequestError("columnName must be provided").build();
+          }
+
+          StargateGrpc.StargateBlockingStub blockingStub =
+              grpcFactory
+                  .constructBlockingStub()
+                  .withCallCredentials(new StargateBearerToken(token));
+
+          Schema.CqlTable table =
+              BridgeSchemaClient.create(blockingStub).findTable(keyspaceName, tableName);
+          if (table.getColumnsList().stream().noneMatch(c -> columnName.equals(c.getName()))) {
+            return jaxrsServiceError(
+                    Response.Status.NOT_FOUND,
+                    String.format("Column '%s' not found in table.", columnName))
```

Comment:
Maybe add name of the table as well?

Reply:
This is the message that was used in v1.

Reply:
Ah ok. Yeah, then let's use that.

Would be good to go over error messages in future but not right now. Thanks!